data/block/block.go

```diff
@@ -11,6 +11,12 @@ import (
 //  necessarily representing a full block body
 type MiniBlockSlice []*MiniBlock
 
+// MiniblockAndHash holds the info related to a miniblock and its hash
+type MiniblockAndHash struct {
```

Comment:
I would leave it as it is as this is a wrapper used in order to optimize calls to hashing function, it would be more work to wrap the miniBlock onbjects afterwards.